tools/mtouch/BuildTasks.mtouch.cs

```diff
@@ -307,50 +307,59 @@ protected override async Task ExecuteAsync ()
+						}
 					}
-				}
-				// mtouch does not validate extra parameters given to GCC when linking (--gcc_flags)
-				if (Target.App.UserGccFlags?.Count > 0)
-					linker_errors.Add (new MonoTouchException (5201, true, Errors.MT5201, StringUtils.FormatArguments (Target.App.UserGccFlags)));
-				else
-					linker_errors.Add (new MonoTouchException (5202, true, Errors.MT5202));
-
-				if (code == 255) {
-					// check command length
-					// getconf ARG_MAX
-					StringBuilder getconf_output = new StringBuilder ();
-					if (Driver.RunCommand ("getconf", new [] { "ARG_MAX" }, output: getconf_output, suppressPrintOnErrors: true) == 0) {
-						int arg_max;
-						if (int.TryParse (getconf_output.ToString ().Trim (' ', '\t', '\n', '\r'), out arg_max)) {
-							var cmd_length = Target.App.CompilerPath.Length + 1 + CompilerFlags.ToString ().Length;
-							if (cmd_length > arg_max) {
-								linker_errors.Add (ErrorHelper.CreateWarning (5217, Errors.MT5217, cmd_length));
+					// mtouch does not validate extra parameters given to GCC when linking (--gcc_flags)
+					if (Target.App.UserGccFlags?.Count > 0)
+						linker_errors.Add (new MonoTouchException (5201, true, Errors.MT5201, StringUtils.FormatArguments (Target.App.UserGccFlags)));
+					else
+						linker_errors.Add (new MonoTouchException (5202, true, Errors.MT5202));
+
+					if (code == 255) {
+						// check command length
+						// getconf ARG_MAX
+						StringBuilder getconf_output = new StringBuilder ();
+						if (Driver.RunCommand ("getconf", new [] { "ARG_MAX" }, output: getconf_output, suppressPrintOnErrors: true) == 0) {
+							int arg_max;
+							if (int.TryParse (getconf_output.ToString ().Trim (' ', '\t', '\n', '\r'), out arg_max)) {
+								if (cmd_length > arg_max) {
+									linker_errors.Add (ErrorHelper.CreateWarning (5217, Errors.MT5217, cmd_length));
+								} else {
+									Driver.Log (3, $"Linker failure is probably not due to command-line length (actual: {cmd_length} limit: {arg_max}");
+								}
 							} else {
-								Driver.Log (3, $"Linker failure is probably not due to command-line length (actual: {cmd_length} limit: {arg_max}");
+								Driver.Log (3, "Failed to parse 'getconf ARG_MAX' output: {0}", getconf_output);
 							}
 						} else {
-							Driver.Log (3, "Failed to parse 'getconf ARG_MAX' output: {0}", getconf_output);
+							Driver.Log (3, "Failed to execute 'getconf ARG_MAX'\n{0}", getconf_output);
 						}
-					} else {
-						Driver.Log (3, "Failed to execute 'getconf ARG_MAX'\n{0}", getconf_output);
 					}
 				}
+			} catch (System.ComponentModel.Win32Exception wex) {
+				/* This means we failed to execute the linker, not that the linker itself returned with a failure */
+				if (wex.NativeErrorCode == 7 /* E2BIG = Too many arguments */ )
+					linker_errors.Add (ErrorHelper.CreateWarning (5217, wex, Errors.MT5217, cmd_length));
+				linker_errors.Add (ErrorHelper.CreateError (5222, wex, Errors.MX5222, wex.Message));
```

Comment:
do we need to emit two errors ?

Reply:
You mean make 5217 an error as well (as opposed to a warning)? It currently says: `Native linking possibly failed because the linker command line was too long ({0} characters).`

Reply:
So I mis-read the `CreateWarning` :) likely because the comment (and message) are referring to an error :)

Still I would expect a single error as
* it might be the only thing (the error) reported in an issue;
* the warning only happens if things fails (it's an error condition); and 
* there's a single problem/error in the build (not two and I hate C++ compilers giving me tons of warnings/errors for a single, often unrelated, cause)